### ISSUE TITLE
[5.x]: `ApplyFieldSaveEvent` has inconsistent `$event->field` value

### ISSUE THREAD

**Issue by lindseydiloreto** [craftcms/cms#16156]
### What happened?

_Sidenote..._ `craft\services\Fields::EVENT_BEFORE_APPLY_FIELD_SAVE` is not included in the [Event Generator](https://craftcms.com/docs/5.x/extend/events.html#event-code-generator).

---

I believe there is a bug in the new [`EVENT_BEFORE_APPLY_FIELD_SAVE`](https://github.com/craftcms/cms/compare/703b601ebae2dc2e4d5196d6dfacd24875f0635b...423aaf777ce7743197dc539be0782713bd0eeee9) event. Specifically, the `$event->field` value is inconsistent based on whether you are:

- (A) Changing the field type in the Control Panel (locally), or
- (B) Applying your Project Config changes.

Here is my [current code](https://github.com/doublesecretagency/craft-googlemaps/blob/d1ea2c7c5743a04b030172a1b33cff0804ff3af9/src/helpers/FieldConversionHelper.php#L172-L193) (shown below via Xdebug).

Locally, (scenario A) the `$event->field` registers as a Google Maps field...

<img width="911" src="https://github.com/user-attachments/assets/3972d27f-532b-4529-9726-dedcb8966ae0">

---

 But if you roll back the database and apply the modified Project Config (scenario B), the `$event->field` registers as an Ether field...

<img width="903" src="https://github.com/user-attachments/assets/87b1e4db-683c-49bb-90d8-d9bee1a997bf">

---

This is tangentially related to a very important question...

⭐️ **What is the best way to determine both the _before_ and _after_ types of the changing field?** ⭐ 

I noticed that `ProjectConfig::EVENT_UPDATE_ITEM` emits a [ConfigEvent](https://github.com/craftcms/cms/blob/d5d42bea20e90ae07c78eedacd6a6238480fdb24/src/events/ConfigEvent.php), which contains both an `$oldValue` and a `$newValue`. Is it possible to introduce something similar to the [ApplyFieldSaveEvent](https://github.com/craftcms/cms/blob/d5d42bea20e90ae07c78eedacd6a6238480fdb24/src/events/ApplyFieldSaveEvent.php)?

### Steps to reproduce

You can get the mostly-functional dev version of this feature here...

```js
"doublesecretagency/craft-googlemaps": "dev-v5-features/address-conversion"
```

1. Install Google Maps & Ether Maps.
2. Create an Ether Map field and populate it with some data.
3. Go to Settings > Fields and find your Map field.
4. **Change the field type to Address (Google Maps) and save.**
5. The field should convert as expected, and existing Address data should be properly ported over.

...however...

6. Roll back the database.
7. **Apply the same change via Project Config.**
8. Notice that these scenarios are handled differently.

### Expected behavior

The `$event->field` value should be consistent whether I am changing the field type initially, or applying that change via Project Config.

Additionally, there should be an easy way to determine what the field type is changing _**to**_ and _**from**_.

### Actual behavior

- Changing the field type via CP sets the `$event->field` to a Google Maps field, but
- Changing the field type via Project Config sets the `$event->field` to an Ether field.

### Craft CMS version

5.5.2

### PHP version

8.2.22

### Operating system and version

Linux 6.11.6-orbstack-00279-g28c6c77332e6

### Database type and version

MySQL 8.0.36

### Image driver and version

Imagick 3.7.0 (ImageMagick 6.9.11-60)

### Installed plugins and versions

- Google Maps v5.0.3 (dev-v5-features/address-conversion)
- Ether Maps v5.0.4

**Comment by i-just** [comment]
The new (after) type is in the `$event->config`. You could get the old (before) type by getting the field record from the database (e.g. by the `$event->field->uid`). That said, the `Fields->applyFieldSave()` method, which triggers the `EVENT_BEFORE_APPLY_FIELD_SAVE` event, already gets that record, so I raised a PR to include the old field config in the event.

**Comment by lindseydiloreto** [comment]
Thanks @i-just, I appreciate you taking a look into this!

I just left a follow-up [comment](https://github.com/craftcms/cms/pull/16168#issuecomment-2489266352) on the PR.

**Comment by brandonkelly** [comment]
Craft 4.13.3 and 5.5.3 are out with the fix in #16168.

**Comment by lindseydiloreto** [comment]
Your latest fixes solved the problem, thanks @i-just and @brandonkelly! 🍺 👏
